tests/rule_based_profiler/parameter_builder/test_parameter_builder.py

```diff
@@ -0,0 +1,485 @@
+)
+def test_parameter_builder_dependencies_evaluated_in_parameter_builder_if_not_precomputed(
+    mock_convert_to_json_serializable: mock.MagicMock,
+    empty_rule_state: Dict[str, Union[Domain, Dict[str, ParameterContainer]]],
+):
+    domain = empty_rule_state["domain"]
+    parameters = empty_rule_state["parameters"]
+
+    my_evaluation_dependency_0_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_0",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+    my_evaluation_dependency_1_parameter_builder: ParameterBuilder = (
+        DummyParameterBuilder(
+            name="my_evaluation_dependency_parameter_name_1",
+            evaluation_parameter_builder_configs=None,
+        )
+    )
+
+    evaluation_parameter_builder_configs: Optional[List[ParameterBuilderConfig]] = [
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_0_parameter_builder.to_json_dict()
+        ),
+        ParameterBuilderConfig(
+            **my_evaluation_dependency_1_parameter_builder.to_json_dict(),
+        ),
+    ]
+    my_dependent_parameter_builder: ParameterBuilder = DummyParameterBuilder(
+        name="my_dependent_parameter_builder",
+        evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+    )
+
+    # Now execute dependent "ParameterBuilder".
+    my_dependent_parameter_builder.build_parameters(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+        batch_request=None,
+        runtime_configuration=None,
+    )
+    assert my_dependent_parameter_builder.call_count == 1
+
+    all_fully_qualified_parameter_names: List[
+        str
+    ] = get_fully_qualified_parameter_names(
+        domain=domain,
+        variables=None,
+        parameters=parameters,
+    )
+
+    """
+    This assertion shows that both "ParameterBuilder" dependencies computed their values, and that dependent
+    "ParameterBuilder" also computed its value (as all names are in list).
+    """
+    assert sorted(all_fully_qualified_parameter_names) == sorted(
+        [
+            "$parameter.raw.my_evaluation_dependency_parameter_name_1",
+            "$parameter.raw.my_evaluation_dependency_parameter_name_0",
+            "$parameter.my_evaluation_dependency_parameter_name_1",
+            "$parameter.my_evaluation_dependency_parameter_name_0",
+            "$parameter.raw.my_dependent_parameter_builder",
+            "$parameter.my_dependent_parameter_builder",
+        ]
+    )
+
+
+# noinspection PyUnresolvedReferences,PyUnusedLocal
+@pytest.mark.unit
+@mock.patch(
+    "great_expectations.rule_based_profiler.parameter_builder.parameter_builder.convert_to_json_serializable",
+    return_value="my_json_string",
+)
+def test_parameter_builder_should_only_evaluate_dependencies_that_are_not_precomputed(
+    mock_convert_to_json_serializable: mock.MagicMock,
```

Comment:
why is this mock required if it's not used?

Reply:
Please see above `https://github.com/great-expectations/great_expectations/pull/8302#discussion_r1265546905` -- happy to describe and see if this is normal or symptomatic of an issue.  Thanks.